src/D2L.Security.OAuth2/Provisioning/IAccessTokenProvider.cs

```diff
@@ -16,15 +15,13 @@ public partial interface IAccessTokenProvider {
 		/// </summary>
 		/// <param name="claims">The set of claims to be included in the token.</param>
 		/// <param name="scopes">The set of scopes to be included in the token.</param>
-		/// <param name="cache">The provided <see cref="ICache"/> does not need to 
 		/// check for token expiration or grace period because the 
 		/// <see cref="IAccessTokenProvider"/> will handle it internally.</param>
 		/// <returns>An access token containing an expiry and the provided claims and scopes.</returns>
 		[GenerateSync]		
 		Task<IAccessToken> ProvisionAccessTokenAsync(
 			IEnumerable<Claim> claims,
-			IEnumerable<Scope> scopes,
-			ICache cache = null
+			IEnumerable<Scope> scopes

```

Comment:
Consolidating on one interface, equivalent to the old non-caching one, and taking the cache inside the factory+constructor.

The LMS cache is instantiated each call but it can be safely cached.